### PR TITLE
chore: remove plugins

### PR DESCRIPTION
**Pull request by tjallingt**
This PR removes the zwave, zigbee, rf, log and oauth2 Homey CLI plugins.
These plugins provided two functions: 
1. installing their associated package in the "preprocess" phase
2. the zwave, zigbee and rf added additional logic when creating a new driver

This change doesn't impact current usage because:
1. We are already running `npm install` before the "preprocess" phase so the first step is only useful when the plugin is newly added, we need to improve the documentation to explain what packages developers need to `npm install`.
2. This functionality is currently broken for the sdk v3 variant of the driver packages, we can fix it or choose to remove it.

I have ideas for a better way to support point 2 if we want it but in the meantime it seems better to remove this.

### CODE REVIEW

**Comment by RobinBol** [comment]
> 2. This functionality is currently broken for the sdk v3 variant of the driver packages, we can fix it or choose to remove it.

What is broken at the moment? I tested it a while back for zwave and zigbeedriver, they install the SDKv3 versions when on SDKv3.

Also, how about the compose plugin? Might be a bit weird to keep around as the only plugin available. We could just check for existence of `.homeycompose` maybe?

File: README.md [hunk 1]
```diff
@@ -17,43 +17,34 @@ $ homey --help
...
+## Useful links
 
-For documentation, refer to [AppPluginZigbee](lib/AppPluginZigbee/index.js).
+### Z-Wave
+The `zwave` installs [homey-meshdriver](https://www.npmjs.com/package/homey-meshdriver).
 
-### RF
-The `rf` plugin installs [homey-rfdriver](https://www.npmjs.com/package/homey-rfdriver), and copies pairing templates to `/.homeycompose/`.
+### Zigbee
+The `zigbee` installs [homey-meshdriver](https://www.npmjs.com/package/homey-meshdriver).
```
Comment (WeeJeWel) [inline README.md:37]: This is outdated. Z-Wave as well.

File: lib/App/index.js [hunk 2]
```diff
@@ -1510,7 +1411,7 @@ class App {
       {
         type: 'confirm',
         name: 'switch_compose',
-        message: 'The Homey compose plugin is not detected. Do you want to use Homey compose? It will split the app.json file into separate files for Drivers, Flow Cards and Discovery Strategies.'
+        message: 'The Homey compose is not detected. Do you want to use Homey compose? It will split the app.json file into separate files for Drivers, Flow Cards and Discovery Strategies.'
```
Comment (WeeJeWel) [inline lib/App/index.js:1414]: This sentence is crooked :p

File: lib/App/index.js [hunk 3]
```diff
@@ -2395,7 +2259,7 @@ class App {
 			{
 				type: 'confirm',
 				name: 'compose',
-				message: 'Use Homey compose plugin?'
+				message: 'Use Homey compose?' // TODO!!!! don't ask, new users don't know what this is
```
Comment (WeeJeWel) [inline lib/App/index.js:2262]: Solve TODO

**Review by WeeJeWel — CHANGES_REQUESTED** [review]
What happens when a user does `homey app create` and gets asked for 'Is this a Z-Wave/Zigbee/RF app?` in both SDK2 and SDK3 scenarios?

File: lib/AppPluginZigbee/index.js [hunk 4]
```diff
@@ -1,90 +0,0 @@
...
-      id: 'homey-meshdriver',
-    });
-
-		await fse.copy(
-			path.join(app.path, 'node_modules', 'homey-meshdriver', 'assets', 'driver', 'zigbee', 'device.js'),
-			path.join(driverPath, 'device.js')
-		);
-		await fse.remove(
-			path.join(driverPath, 'driver.js')
-		);
```
Comment (tjallingt) [inline lib/AppPluginZigbee/index.js:85]: `AppPluginZigbee#createDriver` installs `homey-meshdriver` and copies meshdriver templates

File: lib/AppPluginZwave/index.js [hunk 5]
```diff
@@ -1,251 +0,0 @@
...
-      id: 'homey-meshdriver',
-    });
-
-		await fse.copy(
-			path.join(app.path, 'node_modules', 'homey-meshdriver', 'assets', 'driver', 'zwave', 'device.js'),
-			path.join(driverPath, 'device.js')
-		);
-		await fse.remove(
-			path.join(driverPath, 'driver.js')
-		);
```
Comment (tjallingt) [inline lib/AppPluginZwave/index.js:80]: `AppPluginZwave#createDriver` installs homey-meshdriver and copies meshdriver templates

File: lib/AppPluginRF/index.js [hunk 6]
```diff
@@ -1,230 +0,0 @@
...
-
-		await copyDeviceFile(
-			path.join(app.path, 'node_modules', 'homey-rfdriver', '.compose', 'driver', 'driver.js'),
-			path.join(driverPath, 'driver.js')
-		);
-		await copyDeviceFile(
-			path.join(app.path, 'node_modules', 'homey-rfdriver', '.compose', 'driver', 'device.js'),
-			path.join(driverPath, 'device.js')
-		);
-	}
```
Comment (tjallingt) [inline lib/AppPluginRF/index.js:226]: These templates don't exist anymore in rfdriver v3

**Comment by tjallingt** [comment]
> What happens when a user does `homey app create` and gets asked for 'Is this a Z-Wave/Zigbee/RF app?` in both SDK2 and SDK3 scenarios?

We actually used to ask this when a user runs `homey app driver create`, and I believe that functionality was all subtly broken for SDK v3 apps (for zigbee and zwave it would use meshdriver, rf didn't copy the templates).

If this PR is merged we don't ask these questions and we need to clearly document that users need to run `npm install homey-zigbeedriver`

**Comment by WeeJeWel** [comment]
I suggest we do `npm install homey-zigbeedriver` etc. when people answer Yes, and show an URL to that module's documentation/readme.

**Comment by tjallingt** [comment]
> Also, how about the compose plugin? Might be a bit weird to keep around as the only plugin available. We could just check for existence of .homeycompose maybe?

yea I didn't touch the compose "plugin" code but we don't use `homeyplugins.json` anymore to check if it is enabled, instead we check if there is a `.homeycompose` folder.

**Comment by tjallingt** [comment]
> I suggest we do `npm install homey-zigbeedriver` etc. when people answer Yes, and show an URL to that module's documentation/readme.

I'm worried about asking too many questions to new developers when creating an app. Without the context of what the answer to the question is going to do to your app it can be difficult to answer. So if we were to add it I'd suggest something like: "Should we install homey-zigbeedriver for you? This package provides utilities for a Homey Zigbee app. This is highly recommended when adding support for Zigbee devices".

Alternatively we could add a `homey zigbee driver create` command or something like that

**Comment by WeeJeWel** [comment]
This question isn't asked when creating an app, but when adding a driver!

**Comment by tjallingt** [comment]
Ah sorry, I misunderstood what you meant. I still have the same concerns. I'd prefer cleanly removing for now, I personally don't think its a big issue for the documentation to explain that a user needs to run `npm install homey-zigbeedriver` themselves.

If we ask whether a driver is for a zigbee/zwave/rf device, should fix the templates? (so the question does something useful every time, instead of only the first time)

**Comment by WeeJeWel** [comment]
OK!

File: lib/App/index.js [hunk 7]
```diff
@@ -2452,12 +2311,10 @@ class App {
...
-    // Append the homeycompose dir if used
-    if( answers.compose ) {
-      dirs.push( '.homeycompose' );
-      dirs.push( path.join('.homeycompose', 'flow') );
-      dirs.push( path.join('.homeycompose', 'drivers') );
-    }
+    // Append the homeycompose dir
+    dirs.push( '.homeycompose' );
+    dirs.push( path.join('.homeycompose', 'flow') );
+    dirs.push( path.join('.homeycompose', 'drivers') );
```
Comment (RobinBol) [inline lib/App/index.js:2317]: Unrelated: should we add the other compose folders here as well? Such as `capabilities`, `signals`, `discovery` etc. Only `flow` and `drivers` feels a bit trivial to me.

**Review by RobinBol — APPROVED** [review]

**Comment by tjallingt** [comment]
is that "ok lets merge as is" or "ok i hear your concerns but add the questions to driver create"?

File: lib/App/index.js (hunk 7 above)
Comment (WeeJeWel) [reply lib/App/index.js:2317]: Sounds good!

**Comment by WeeJeWel** [comment]
> is that "ok lets merge as is" or "ok i hear your concerns but add the questions to driver create"?

Haha sorry. I meant 'OK, let's remove it'.

File: lib/App/index.js (hunk 7 above)
Comment (RobinBol) [reply lib/App/index.js:2317]: See: https://github.com/athombv/node-homey/pull/137

**Pull request merged**